hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/block/DeletedBlockLogImpl.java

```diff
@@ -426,6 +426,11 @@ public void onMessage(
 
     DatanodeDetails details = deleteBlockStatus.getDatanodeDetails();
     UUID dnId = details.getUuid();
+    UUID dnIdLocal = UUID.fromString(dnId.toString());
+    if (LOG.isDebugEnabled()) {
+      LOG.debug("remoteId = {}, localDnId = {}, remoteDnId == localDnId[{}]",
+          dnId, dnIdLocal, (dnId == dnIdLocal));
```

Comment:
@adoroszlai Thank you very much for your reply! Your comment on object comparison is correct; I should use the equal method for comparison.

> Is it possible that it can proceed simply because SCM was restarted?

I still have some questions about this issue. Later, I will provide a 7-day Grafana monitoring report. The problem we're facing is that while SCM allows deletions, the deletion queue fails to decrease after a period of time. Restarting SCM temporarily restores normal deletion operations, but then the deletion queue stops decreasing again. However, since applying this PR, deletions have been ongoing for 4 days now. So, I might have mistakenly attributed this PR's effect, but there could be other issues at play.

Reply:
Sorry, the response is a bit late. The following image shows our SCM's 7-day deletion monitoring. From the monitoring data, SCM can delete blocks, but there are long periods of stalling in between. During this period, numerous 'UnkowDataNode' WARNING alerts are reported.

![image](https://github.com/user-attachments/assets/95544552-a1a7-4040-acd5-3adfbd2f793f)